codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala

```diff
@@ -0,0 +1,332 @@
+    if (commands.exists(_.hasStream))
+      "akka.NotUsed" :: "akka.stream.scaladsl.Source" :: Nil
+    else
+      Nil
+  }
+
+  private[codegen] def actionSource(service: ModelBuilder.ActionService, packageName: String): String = {
+
+    val className = service.className
+
+    val imports = generateImports(
+      service.commandTypes,
+      packageName,
+      otherImports = Seq(
+        "com.akkaserverless.scalasdk.action.Action",
+        "com.akkaserverless.scalasdk.action.ActionCreationContext") ++ streamImports(service.commands),
+      semi = false)
+
+    val methods = service.commands.map { cmd =>
+      val methodName = cmd.name
+      val input = lowerFirst(cmd.inputType.name)
+      val inputTypeFullName = cmd.inputType.fullName
+      val outputType = cmd.outputType.fullName
+
+      if (cmd.isUnary) {
+        val jsonTopicHint = {
+          // note: the somewhat funky indenting is on purpose to lf+indent only if comment present
+          if (cmd.inFromTopic && cmd.inputType.fullQualifiedName == "com.google.protobuf.Any")
+            """|// JSON input from a topic can be decoded using JsonSupport.decodeJson(classOf[MyClass], any)
+               |  """.stripMargin
+          else if (cmd.outToTopic && cmd.outputType.fullQualifiedName == "com.google.protobuf.Any")
+            """|// JSON output to emit to a topic can be encoded using JsonSupport.encodeJson(myPojo)
+               |  """.stripMargin
+          else ""
+        }
+
+        s"""|/** Handler for "$methodName". */
+            |override def ${lowerFirst(methodName)}($input: $inputTypeFullName): Action.Effect[$outputType] = {
+            |  ${jsonTopicHint}throw new RuntimeException("The command handler for `$methodName` is not implemented, yet")
+            |}""".stripMargin
+      } else if (cmd.isStreamOut) {
+        s"""
+           |/** Handler for "$methodName". */
+           |override def ${lowerFirst(methodName)}($input: $inputTypeFullName): Source[Action.Effect[$outputType], NotUsed] = {
+           |  throw new RuntimeException("The command handler for `$methodName` is not implemented, yet")
+           |}""".stripMargin
+      } else if (cmd.isStreamIn) {
+        s"""
+           |/** Handler for "$methodName". */
+           |override def ${lowerFirst(methodName)}(${input}Src: Source[$inputTypeFullName, NotUsed]): Action.Effect[$outputType] = {
+           |  throw new RuntimeException("The command handler for `$methodName` is not implemented, yet")
+           |}""".stripMargin
+      } else {
+        s"""
+           |/** Handler for "$methodName". */
+           |override def ${lowerFirst(methodName)}(${input}Src: Source[$inputTypeFullName, NotUsed]): Source[Action.Effect[$outputType], NotUsed] = {
+           |  throw new RuntimeException("The command handler for `$methodName` is not implemented, yet")
+           |}""".stripMargin
+      }
+    }
+
+    s"""|$unmanagedComment
+        |
+        |package $packageName
```

Comment:
in latest ValueEntity and View generation we removed the `packageName` parameter and use `${view.fqn.parent.scalaPackage}` here